File: app/src/main/java/com/example/dkalev/remember/model/Card.java
package com.example.dkalev.remember.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;
import android.graphics.Bitmap;

import java.sql.Blob;

@Entity(foreignKeys = @ForeignKey(
        entity = Deck.class,
        parentColumns = "deck_id",
        childColumns = "deck_id"))
public class Card {

    @PrimaryKey(autoGenerate = true)
    private int uid;

    @ColumnInfo(name = "text_front")
    private String textFront;

    @ColumnInfo(name = "text_back")
    private String textBack;

    @ColumnInfo(name = "deck_id")
    private int deckId;

    public Card(int deckId){
        this.deckId = deckId;
    }


//    public void setImageFront(Bitmap imageFront) {
//        this.imageFront = imageFront;
//    }
//
//    public void setImageBack(Bitmap imageBack) {
//        this.imageBack = imageBack;
//    }

//    @ColumnInfo(name = "image_front")
//
//    private Bitmap imageFront;
//
//    @ColumnInfo(name = "image_back")
//    private Bitmap imageBack;

    public void setUid(int uid) {
        this.uid = uid;
    }

    public int getUid() {
        return uid;
    }

    public String getTextFront() {
        return textFront;
    }

    public String getTextBack() {
        return textBack;
    }

    public void setTextFront(String textFront) {
        this.textFront = textFront;
    }

    public void setTextBack(String textBack) {
        this.textBack = textBack;
    }

    public int getDeckId() {
        return deckId;
    }

    public void setDeckId(int deckId) {
        this.deckId = deckId;
    }
}

File: app/src/main/java/com/example/dkalev/remember/model/card/Card.kt
package com.example.dkalev.remember.model.card

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey
import android.arch.persistence.room.ForeignKey.CASCADE
import android.arch.persistence.room.PrimaryKey
import com.example.dkalev.remember.model.deck.Deck
import java.util.*

@Entity(foreignKeys = [ForeignKey(
        entity = Deck::class,
        parentColumns = arrayOf("deck_name"),
        childColumns = arrayOf("deck_name"),
        onDelete = CASCADE)])//onDelete = CASCADE -> delete all cards when deleting the deck
data class Card(@field:ColumnInfo(name = "deck_name")
           var deckName: String?) {

    //    public void setImageFront(Bitmap imageFront) {
    //        this.imageFront = imageFront;
    //    }
    //
    //    public void setImageBack(Bitmap imageBack) {
    //        this.imageBack = imageBack;
    //    }

    //    @ColumnInfo(name = "image_front")
    //
    //    private Bitmap imageFront;
    //
    //    @ColumnInfo(name = "image_back")
    //    private Bitmap imageBack;

    @PrimaryKey(autoGenerate = true)
    var uid: Int = 0

    @ColumnInfo(name = "text_front")
    var textFront: String? = null

    @ColumnInfo(name = "text_back")
    var textBack: String? = null

    @ColumnInfo(name = "difficulty")
    var difficulty: Double = 0.3

    @ColumnInfo(name = "days_between_reviews")
    var daysBetweenReviews: Int = 0

    @ColumnInfo(name = "day_last_reviewed")
    var dayLastReviewed: Date? = null

    @ColumnInfo(name = "due_date")
    var dueDate: Date? = null
}